gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java

```diff
@@ -519,6 +528,12 @@ public boolean isQueryParsingSuccessful()
         return isQueryParsingSuccessful;
     }
 
+    @JsonProperty
+    public Optional<String> getParseErrorMessage()
```

Comment:
This method is used only from tests. What's the motivation of this change?

Reply:
We have server connected with gateway that runs explain before actually running the query. 
But not all queries need to be sent with explain.

For example there are cases where queries with parsing failure (no need to run explain cause it is highly likely to fail anyway due to syntax error)
  - we need to tell user on WHY query has failed, but running another sql parser or explain is waste of resource
  - since gateway already does query parsing, why not use that failure info.

Reply:
> since gateway already does query parsing, why not use that failure info.

I don't think relying on Trino Gateway's parser is safe. For instance, the behavior might be different from the backend Trino because of version mismatch.

Reply:
Sure updated description.

> I don't think relying on Trino Gateway's parser is safe. For instance, the behavior might be different from the backend Trino because of version mismatch.

That is true. Gateway does not guarantee that, but I think we should leave this up to the users to decide on how much they rely on the result. 
Btw in our case, we just use `isQueryParsingSuccessful` to see if we will run explain to fetch IO Plan or not, as it is part of our routing rules. (There are several steps on deciding which routing group to route, but one heavy step of running explain can be skipped). 
This failure info is a supplementary info where we tell users that query parsing has failed so please do check your sql 
- As of now, we use sql parser libraries to see why parsing has failed, which is also not that accurate.